src/ng/directive/select.js

```diff
@@ -116,9 +120,26 @@ var SelectController =
 
   self.registerOption = function(optionScope, optionElement, optionAttrs, interpolateValueFn, interpolateTextFn) {
 
-    if (interpolateValueFn) {
+    if (interpolateValueFn === true) {
```

Comment:
Or perhaps we don't need to do that at all, we could just check whether `optionAttrs.ngValue` is defined?